src/Common/src/System/Security/Cryptography/Asn1/AsnXml.targets

```diff
@@ -0,0 +1,43 @@
+<Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <AsnXml />
+  </PropertyGroup>
+  <ItemGroup>
+    <None Include="$(MSBuildThisFileDirectory)asn.xsd">
+      <Link>Common\System\Security\Cryptography\Asn1\asn.xsd</Link>
+    </None>
+  </ItemGroup>
+
+  <!-- MSBuild for .NET Core doesn't support XslTransform at this time -->
+  <Target Name="CompileAsn" BeforeTargets="CoreCompile"
+    Condition="'$(RunningOnCore)' != 'true'"
+    Inputs="@(AsnXml);$(MSBuildThisFileDirectory)asn.xslt"
+    Outputs="%(Identity).cs">
+
+    <MakeDir Directories="$(IntermediateOutputPath)asnxml" />
+
+    <XslTransformation
+      XslInputPath="$(MSBuildThisFileDirectory)asn.xslt"
+      XmlInputPaths="@(AsnXml)"
+      OutputPaths="@(AsnXml -> '$(IntermediateOutputPath)asnxml\%(filename).cs')" />
```

Comment:
JFYI I was building this from VS4Mac with .NET Core 2.1 target and it worked. Does that use some other msbuild than the .NET Core one?

Reply:
Apparently it uses MSBuild on Mono instead of the .NET Core version of MSBuild.

The .NET Core build doesn't have XslTransformation.  https://github.com/Microsoft/msbuild/issues/3368